core/src/main/java/org/mule/runtime/core/routing/requestreply/AbstractAsyncRequestReplyRequester.java

```diff
@@ -128,8 +125,8 @@ public void setReplySource(MessageSource messageSource) {
 
   @Override
   public void initialise() throws InitialisationException {
-    name = format(NAME_TEMPLATE, storePrefix, muleContext.getConfiguration().getId(),
-                  flowConstruct == null ? "" : flowConstruct.getName());
+    name =
+        format(NAME_TEMPLATE, storePrefix, muleContext.getConfiguration().getId(), getLocation().getParts().get(0).getPartPath());
```

Comment:
Refactor repeated use of this type of thing into static helpers in ComponentLocation.